controller_manager/src/controller_manager.cpp

```diff
@@ -869,9 +879,20 @@ controller_interface::return_type ControllerManager::switch_controller(
         command_interface_names.end());
     };
 
+    const auto extract_command_data_for_controller =
+      [this](const ControllerSpec controller, std::vector<std::string> & request_data_list)
```

Comment:
Does anything need to be captured in this lambda?

Reply:
I don't have much experience with lambdas to be honest, and I was just following the style found with lambda functions nearby in the code.

That said, the only outside variable I see that we are using is 'controller', so maybe it should just be [&]?

Reply:
So we should be able to get away without capturing anything in this lambda, because you're only using the two function arguments `controller` and `request_data_list`, which you're passing in when you call the lambda! (hence they need not be captured)

```suggestion
      [](const ControllerSpec controller, std::vector<std::string> & request_data_list)
```

Reply:
Good point! I committed that change